internal/federation/server.go

```diff
@@ -243,11 +243,42 @@ func (s *Server) SendFederationRequest(
 	if httpError, ok := err.(gomatrix.HTTPError); ok {
 		t.Logf("[SSAPI] %s %s%s => error(%d): %s (%s)", req.Method(), req.Destination(), req.RequestURI(), httpError.Code, err, time.Since(start))
 	} else if err == nil {
-		t.Logf("[SSAPI] %s %s%s => 2xx (%s)", req.Method(), req.Destination(), req.RequestURI(), time.Since(start))
```

Comment:
While working with `DoFederationRequest`, I noticed this will unconditionally insert this `2xx` while not checking if it's the actual status code, as the test writer might expect an error, adjust `resBody` to the error body, receive an expected `4xx`, and this'll still say `2xx`.

There's no way to fetch the actual status code at this point, as the response body is already cleaned up in the `DoRequestAndParseResponse` function.

Reply:
Are you sure? The docs in GMSL state:
```go
// DoRequestAndParseResponse calls DoHTTPRequest and then decodes the response.
//
// If the HTTP response is not a 200, an attempt is made to parse the response
// body into a gomatrix.RespError. In any case, a non-200 response will result
// in a gomatrix.HTTPError.
func (fc *Client) DoRequestAndParseResponse(
```

Reply:
Oh! In that case I'll revert this, thanks.